core-rust/state-manager/src/types.rs

```diff
@@ -480,10 +480,11 @@ pub enum CommitError {
     MissingEpochProof,
 }
 
-#[derive(Debug, Decode, Encode, Categorize)]
+#[derive(Debug, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
```

Comment:
We should be able to just use `Decode`/`Encode`/`Categorize` here - (and maybe `StateHash` if necessary?).

ScryptoX is only needed for types which require the Scrypto SBOR custom value model (ie things referencing runtime engine types).

Reply:
ahh, indeed ✅